web/src/main/java/com/microsoft/applicationinsights/web/internal/WebRequestTrackingFilter.java

```diff
@@ -284,6 +301,38 @@ private String getName(ServletContext context) {
         return name;
     }
 
+    /*This method is used to parse application name from the properties file*/
+    private String getApplicatioNameFromProperties() {
+
+        Properties properties = new Properties();
+        InputStream input = null;
+        String name = null;
+        try {
+            String fileName = "application.properties";
+            input = WebRequestTrackingFilter.class.getClassLoader().getResourceAsStream(fileName);
+            if (input == null) {
+                return null;
+            }
+
+            properties.load(input);
+            name = properties.getProperty("spring.application.name");
```

Comment:
Is it the way we're going to use for spring only or the name should be more generic?

Reply:
@Dmitry-Matveev yes as of now I am not aware of any other framework that reads the value from the properties file though this is through a very preliminary research. There can be places where this is the only way but not in my knowledge. For all other cases we are already reading from application context. Also just pushed the corrected version where the public constructor has been removed which was accidentally present in the previous commit.

Reply:
This property is going to be our public interface to configure application name that something can take dependency on, so changing it once released is not an easy task.
I think renaming this property before release to a more generic scope rather than "spring" will make us stay on the safe side and prevent "deprecation"/rename issues in the future.